src/alert/alert.ts

```diff
@@ -38,12 +43,22 @@ export class NgbAlert {
   /**
    * An event emitted when the close button is clicked. This event has no payload. Only relevant for dismissible alerts.
    */
-  @Output() close = new EventEmitter();
+  @Output() close = new EventEmitter<void>();
 
-  constructor(config: NgbAlertConfig) {
+  constructor(config: NgbAlertConfig, private _renderer: Renderer2, private _element: ElementRef) {
     this.dismissible = config.dismissible;
     this.type = config.type;
   }
 
   closeHandler() { this.close.emit(null); }
+
+  ngOnChanges(changes: SimpleChanges) {
+    const {type} = changes;
+    if (type && !type.firstChange) {
+      this._renderer.removeClass(this._element.nativeElement, `alert-${type.previousValue}`);
+      this._renderer.addClass(this._element.nativeElement, `alert-${type.currentValue}`);
+    }
+  }
+
+  ngOnInit() { this._renderer.addClass(this._element.nativeElement, `alert-${this.type}`); }
```

Comment:
Couldn't we move this logic to `ngOnChanges` and set the class on `firstChange`? This way the whole logic would be centralised in one place